File: cpp/src/Point.cpp
#include "Point.h"

//
// Created by alon on 15/11/16.
//
using namespace std;

/**
 * the c-tor of the Point.
 * @param x1
 * @param y2
 * @return nothing.
 */
Point::Point(int x1, int y2) {
    x = x1;
    y = y2;
}

/**
 * the second c-tor of the Point.
 * @param p1
 * @return nothing.
 */
Point::Point(Point *p1) {
    x = p1->getX();
    y = p1->getY();
}

/**
 * return the x value of the Point.
 * @return x.
 */
int Point::getX() const {
    return this->x;
}

/**
 * return the y value of the Point.
 * @return y.
 */
int Point::getY() const {
    return this->y;
}

/**
 * operator for check of 2 points are equal.
 * @param point2.
 * @return bool.
 */
bool Point::operator==(const Point &point2) {
    return (x == point2.getX() && y == point2.getY());
}

/**
 * operator for print a point.
 * @param os.
 * @param p.
 * @return the x and y of the point.
 */
ostream &operator<<(std::ostream &os, const Point &p) {
    return os << "(" << p.x << "," << p.y << ")";
}

Point::Point() {
    this->x=0;
    this->y=0;
}

File: java/Point.java
/**
 * Class representing a point on the Map, which has a x-value and a y-value.
 */
public class Point {
    private int x;
    private int y;

    /**
     * Default Ctor, initialize point as (0,0).
     */
    public Point() {
        this(0, 0);
    }

    /**
     * Initializes the point as (x,y).
     *
     * @param x x-value of point.
     * @param y y-value of point.
     */
    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    /**
     * @return the x value of the point.
     */
    public int getX() {
        return x;
    }

    /**
     * @return the y value of the point.
     */
    public int getY() {
        return y;
    }

    /**
     * Convenience method.
     *
     * @param x x-value to be checked.
     * @param y y-value to be checked.
     * @return true if the values are equal to the values of the point, otherwise false.
     */
    public boolean isEqual(int x, int y) {
        return this.x == x && this.y == y;
    }

    /**
     * @param other other point to compare.
     * @return true if the points are equal, otherwise false.
     */
    public boolean isEqual(Point other) {
        return isEqual(other.getX(), other.getY());
    }
}
